geo/geopoint.go

```diff
@@ -10,16 +10,35 @@ import (
 var polygon = "Polygon"
 
 const (
-	twoPi         float64 = 2 * math.Pi
-	radiusOfEarth float64 = 6378137 //(in metres defined by wgs84)
+	twoPi                  float64 = 2 * math.Pi
+	radiusOfEarth          float64 = 6378137 //(in metres defined by wgs84)
+	defaultConcurrency             = 10      // limit number of go routines to not put too much on heap
+	defaultMaximumSegments         = 180
 )
 
-type maxSegments struct {
-	mu       sync.Mutex
-	segments int
+// Config object to define geo configurations
+type Config struct {
+	defaultConcurrencyLimit int
+	defaultMaxSegments      int
 }
 
-var defaultMaxSegments = maxSegments{segments: 180}
+// Default geo configuration for methods on config receiver
+var DefaultConfig = &Config{
+	defaultConcurrencyLimit: defaultConcurrency,
+	defaultMaxSegments:      defaultMaximumSegments,
+}
+
+// List of errors
+var (
+	ErrRadiusLargerThanEarth    = errors.New("radius can not be greater than the radius of the Earth, 6378137 metres")
```

Comment:
Line 33 seems a little extreme! Are DP using this repo to measure areas across the whole Earth or just the UK? If just the UK then maybe half the length of the country should be the maximum radius. According to Google the length of the UK is 965km https://www.nationsencyclopedia.com/Europe/United-Kingdom-LOCATION-SIZE-AND-EXTENT.html. So I guess half is about 482.5km or 482500 metres.

Reply:
Remember this is a library open to the public, so anyone could be using it across the globe

Reply:
Following from the above, one may think it silly to generate a circular boundary anywhere on the globe that is exceedingly large but how would we know that there isn't a user case for this.

Fyi - I mostly copied this code by converting the same functions from another language into Go for our use (I couldn't find an existing Go library)

Reply:
To be honest I'm not sure who will be using this library so if there's a user case then fair enough.